File: csharp/Stack.cs
using System;
using System.Collections.Generic;

public class MyStack<T>
{
    private List<T> list = new List<T>();

    public void Push(T item) => list.Add(item);
    public T Pop()
    {
        if (list.Count == 0) return default;
        var item = list[^1];
        list.RemoveAt(list.Count - 1);
        return item;
    }

    public T Peek() => list.Count > 0 ? list[^1] : default;
    public void Print() => Console.WriteLine(string.Join(", ", list));
}

public static class StackDemo
{
    public static void RunStackDemo()
    {
        Console.WriteLine("🔁 Stack Demo");
        var stack = new MyStack<int>();
        stack.Push(1);
        stack.Push(2);
        stack.Push(3);
        stack.Print();
        Console.WriteLine("Popped: " + stack.Pop());
        Console.WriteLine("Peek: " + stack.Peek());
        stack.Print();
    }
}

File: python/stack.py
class MyStack:
    def __init__(self):
        self.stack = []

    def push(self, x):
        self.stack.append(x)

    def pop(self):
        return self.stack.pop() if self.stack else None

    def peek(self):
        return self.stack[-1] if self.stack else None

    def print_stack(self):
        print(self.stack)
